WordPress/gutenberg
# Add home template details to inspector controls

## What?
Part of: https://github.com/WordPress/gutenberg/issues/59689

Designs are from: https://github.com/WordPress/gutenberg/issues/59689#issuecomment-2049394858

This PR adds the  home template details to inspector controls, similar to the ones that exist in details sidebar.

## Testing Instructions
1. In site editor go to edit the home or index template and observe the `Blog title, Posts per page and Discussion` rows in inspector controls.


## Screenshots or screencast 
<img width="631" alt="Screenshot 2024-05-20 at 1 27 53 PM" src="https://github.com/WordPress/gutenberg/assets/16275880/04f284f5-1e80-4e30-b734-52cec777c7bf">
